docs/src/model_simulation/examples/periodic_events_simulation.md

```diff
@@ -0,0 +1,95 @@
+
+## [Modelling a circadian periodic event in an ODE simulation](@id periodic_event_simulation_example_ode)
+We will consider a simple circadian model, consisting of a single protein ($X$), which is phosphorylated ($X \to Xᴾ$) in the presence of light ($l$). Here, the light parameter can either be $0$ (night) or $1$ (day). We can model this using a simple periodic event which switches the value of $l$ every 12 hours (here, `%` is the [remainder operator](https://docs.julialang.org/en/v1/manual/mathematical-operations/#Arithmetic-Operators)).
+```@example periodic_event_example
+using Catalyst
+circadian_model = @reaction_network begin
+    @discrete_events begin
+        12 => [l ~ (l + 1)%2]
+    end
+ (kₚ*l,kᵢ), X <--> Xᴾ
+end
+```
+We can now simulate this model, observing how a 24-hour cycle is reached
+```@example periodic_event_example
+using OrdinaryDiffEq, Plots
+u0 = [:X => 150.0, :Xᴾ => 50.0]
+ps = [:kₚ => 0.1, :kᵢ => 0.1, :l => 1.0]
+oprob = ODEProblem(circadian_model, u0, (0.0, 100.0), ps)
+sol = solve(oprob)
+plot(sol)
+```
+
+## [Modelling a circadian periodic event in a jump simulation](@id periodic_event_simulation_example_ode)
+While defining periodic events is easy for ODE and SDE simulations, due to how events are internally implemented, these cannot currently be used for jump simulations. Instead, we create a new model (and its corresponding `JumpProblem`), without the event:
+```@example periodic_event_example
+circadian_model = @reaction_network begin
+ (kₚ*l,kᵢ), X <--> Xᴾ
+end
+
+using JumpProcesses
+u0 = [:X => 150, :Xᴾ => 50]
+ps = [:kₚ => 0.1, :kᵢ => 0.1, :l => 1.0]
+dprob = DiscreteProblem(circadian_model, u0, (0.0, 100.0), ps)
+jprob = JumpProblem(circadian_model, dprob, Direct())
+nothing # hide
+```
+Next, we implement our event through a `DiscreteCallback`. It triggers every 12 time units, and when it does, it flips the value of `l`. Since we will interface with $l$'s value multiple times, we [create specific `getl` and `setl` functions](@ref simulation_structure_interfacing_functions) to do this (in practice, due to this model's small size, the performance impact is negliable).
+```@example periodic_event_example
+getl = ModelingToolkit.getp(jprob, :l)
+setl = ModelingToolkit.setp(jprob, :l)
+function condition(u, t, integrator)
+ (t % 12) == 0.0
+end
+function affect!(integrator)
+    setl(integrator, (getl(integrator) + 1) % 2)
+    reset_aggregated_jumps!(integrator)
+end
+callback = DiscreteCallback(condition, affect!)
+nothing # hide
+```
+!!! danger
+    Whenever the integrator of a jump simulator is modified, the `reset_aggregated_jumps!` function must be called (with the integrator as its input) afterwards. Omitting to do so will result in erroneous simulations.
```

Comment:
```suggestion
    Whenever the integrator of a jump simulator is modified, the `reset_aggregated_jumps!` function must be called (with the integrator as its input) afterwards. Omitting to do so will result in simulations (incorrectly) using the old values of parameters/states internally instead of the updated values.
```